kernel/src/scan/data_skipping.rs

```diff
@@ -75,6 +75,36 @@ impl DataSkippingFilter {
         let (predicate, referenced_schema) = physical_predicate?;
         debug!("Creating a data skipping filter for {:#?}", predicate);
 
+        // Convert all fields into nullable, as stats may not be available for all columns
+        // (and usually aren't for partition columns).
+        struct NullableStatsTransform;
+        impl<'a> SchemaTransform<'a> for NullableStatsTransform {
+            fn transform_struct_field(
+                &mut self,
+                field: &'a StructField,
+            ) -> Option<Cow<'a, StructField>> {
+                use Cow::*;
+                let field = match self.transform(&field.data_type)? {
+                    Borrowed(data_type) => match field.is_nullable() {
+                        true => Borrowed(field),
+                        false => Owned(StructField {
+                            name: field.name.clone(),
+                            data_type: data_type.clone(),
+                            nullable: true,
+                            metadata: field.metadata.clone(),
+                        }),
+                    },
+                    Owned(data_type) => Owned(StructField {
+                        name: field.name.clone(),
+                        data_type,
+                        nullable: true,
+                        metadata: field.metadata.clone(),
+                    }),
+                };
```

Comment:
There's a fair amount of redundancy here, I wonder if we can do something more streamlined?
```suggestion
                let field = match self.transform(&field.data_type)? {
                    Borrowed(data_type) if field.is_nullable() => Borrowed(field),
                    data_type => Owned(StructField {
                        name: field.name.clone(),
                        data_type: data_type.into_owned(),
                        nullable: true,
                        metadata: field.metadata.clone(),
                    }),
                };
```

Reply:
Good point, that's much nicer thanks.